analyzer/codechecker_analyzer/analyzers/analyzer_types.py

```diff
@@ -184,7 +202,18 @@ def check_supported_analyzers(analyzers):
         # Check version compatibility of the analyzer binary.
         if analyzer_bin:
             analyzer = supported_analyzers[analyzer_name]
-            if not analyzer.version_compatible(analyzer_bin, check_env):
+            if requested_version:
+                bin_version = StrictVersion(str(analyzer.version_info()))
```

Comment:
In case of ClangSA the `analyzer.version_info()` returns a `ClangVersionInfo` object which is a custom object. This doesn't convert to a string that could be fed by `StrictVersion`. If `ClangSA.version_info()` would also return a `StrictVersion` then no conversion would be needed.